### ISSUE TITLE
Cannot Marshal math/big.Int

### ISSUE THREAD

**Issue by jeffmitchell-sas** [amazon-ion/ion-go#193]
Hello, I think I have come across an issue while trying to marshal a math/big.Int, 18446744073709551615 using ion.MarshalText(). Sample code is below. 

ion.MarshalText() returns "{}" for big.Int but it seems to handle it just fine for int64 values, i.e. 9223372036854775807.

I am unsure if this is on purpose and I apologize if I am doing something wrong. 

``` go
package main

import (
	"fmt"
	"math/big"

	"github.com/amazon-ion/ion-go/ion"
)

func main() {
	// big int
	bigInt := new(big.Int).Neg(new(big.Int).SetUint64(18446744073709551615))
	a, err := ion.MarshalText(bigInt)
	if err != nil {
		panic(err)
	}
	fmt.Println(string(a)) // prints "{}"

	// int64
	var myInt int = 9223372036854775807
	var myInt64 int64 = int64(myInt)
	b, err := ion.MarshalText(myInt64)
	if err != nil {
		panic(err)
	}
	fmt.Println(string(b))  // prints 9223372036854775807
}
```

**Comment by tgregg** [comment]
Thanks, @jeffmitchell-sas. It looks like the implementation isn't wired up yet to support writing big.Int values via MarshalText. See https://github.com/amazon-ion/ion-go/blob/master/ion/marshal.go#L212

However I do see that the writer has a `writeBigInt` function: https://github.com/amazon-ion/ion-go/blob/master/ion/textwriter.go#L97

Fixing this issue may be as simple as updating the `encodeValue` function linked above to match on `big.Int` and call through to `writeBigInt` in that case. This could be a great first issue for a new contributor.

In the meantime, it may be possible to work around this problem by using a text writer directly, sidestepping the `MarshalText` function.

**Comment by jeffmitchell-sas** [comment]
Thanks for your reply @tgregg. I'd like to contribute if there is an opportunity. Do you have any guidelines written up?

**Comment by popematt** [comment]
There are some general contribution guidelines in the `CONTRIBUTING.md` file and a development section in the [README](https://github.com/amazon-ion/ion-go?tab=readme-ov-file#development).

We don't have any special code style (that I'm aware of) or other special requirements. If you can write reasonably idiomatic Go, then you already know pretty much everything you need to contribute to this repo.

**Comment by jeffmitchell-sas** [comment]
@popematt, I might not have the correct access yet to create a branch/PR so I am attaching a patch here on my fix. I added a tests in mashal_test.go and it looks to be ok. Please let me know if this looks good otherwise or if there are any comments. 
[issue_193.diff.zip](https://github.com/amazon-ion/ion-go/files/15120055/issue_193.diff.zip)

**Comment by jeffmitchell-sas** [comment]
Oh, I see that I have to fork the project. Will do that.

**Comment by jeffmitchell-sas** [comment]
Fixed in [v1.4.0](https://github.com/amazon-ion/ion-go/releases/tag/v1.4.0)
